src/main/java/cn/nukkit/entity/passive/EntityVillager.java

```diff
@@ -1,13 +1,27 @@
 package cn.nukkit.entity.passive;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import cn.nukkit.Player;
+import cn.nukkit.entity.Entity;
 import cn.nukkit.entity.EntityAgeable;
 import cn.nukkit.entity.EntityCreature;
+import cn.nukkit.inventory.Inventory;
+import cn.nukkit.inventory.InventoryHolder;
+import cn.nukkit.inventory.TradeInventory;
+import cn.nukkit.inventory.TradeInventoryRecipe;
+import cn.nukkit.item.Item;
 import cn.nukkit.level.format.FullChunk;
 import cn.nukkit.nbt.tag.CompoundTag;
+import cn.nukkit.nbt.tag.ListTag;
 
-public class EntityVillager extends EntityCreature implements EntityNPC, EntityAgeable {
+public class EntityVillager extends EntityCreature implements InventoryHolder, EntityNPC, EntityAgeable {
```

Comment:
Not sure about implementing InventoryHolder here, since it has no physical inventory

Reply:
or does it?

Reply:
Villager trading is inventory and since villager contains trade recipes, i find them as inventory holders.